### PR TITLE
[Revalidation] Add configurable overrides for revalidation rates

### PR DESCRIPTION
**Pull request by loic-sharma**
This will let us speed up/slow down the revalidation job by changing Octopus variables and creating a new deployment.

Addresses https://github.com/NuGet/Engineering/issues/1789

### CODE REVIEW

File: src/NuGet.Services.Revalidate/Job.cs [hunk 1]
```diff
@@ -112,6 +119,12 @@ await scope.ServiceProvider
         protected override void ConfigureJobServices(IServiceCollection services, IConfigurationRoot configurationRoot)
         {
             services.Configure<RevalidationConfiguration>(configurationRoot.GetSection(JobConfigurationSectionName));
+            services.Configure<RevalidationConfiguration>(config =>
```
Comment (loic-sharma) [inline src/NuGet.Services.Revalidate/Job.cs]: Yes, I'm configuring the same configuration twice. See [this documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-2.1#configure-simple-options-with-a-delegate):

> You can add multiple configuration providers... They're applied in the order that they're registered... When more than one configuration service is enabled, the last configuration source specified wins and sets the configuration value.

Comment (agr) [reply src/NuGet.Services.Revalidate/Job.cs]: Shouldn't you then have 
```
services.Configure<RevalidationConfiguration>(config =>
{
    config.MinPackageEventRate = _overrideMinPackageEventRate;
    config.MaxPackageEventRate = _overrideMaxPackageEventRate;
});
```
as the first `Configure` call, followed by
```
services.Configure<RevalidationConfiguration>(configurationRoot.GetSection(JobConfigurationSectionName));
```
Which will override defaults if specified?

Comment (loic-sharma) [reply src/NuGet.Services.Revalidate/Job.cs]: The overrides should override the configuration. The `Configure` methods are called in the order that they're registered:

1. The setting's files configs are applied
2. The CLI overrides are applied on top of the setting's files

I've deployed this and verified the behavior is as expected.

File: src/NuGet.Services.Revalidate/Scripts/NuGet.Services.Revalidate.cmd [hunk 2]
```diff
@@ -9,6 +9,8 @@ title #{Jobs.nuget.services.revalidate.Title}
 
 start /w NuGet.Services.Revalidate.exe ^
     -Configuration #{Jobs.nuget.services.revalidate.Configuration} ^
+    -OverrideMinPackageEventRate #{Jobs.nuget.services.revalidation.MinPackageEventRate} ^
```
Comment (joelverhagen) [inline src/NuGet.Services.Revalidate/Scripts/NuGet.Services.Revalidate.cmd]: Let's move this to the JSON. As talked about offline, we have other Octopus replacement tokens in the JSON so we can't just mimic that.

**Pull request merged**